modules/loader.js

```diff
@@ -846,90 +842,87 @@ __private.snapshotFinished = err => {
  * @todo Add description for the params
  */
 __private.loadBlocksFromNetwork = function(cb) {
-	let errorCount = 0;
+	// Number of failed attempts to load from peers.
+	let failedAttemptsToLoad = 0;
+	// If True, own node's db contains same number of blocks than the asked peer.
 	let loaded = false;
 
 	async.whilst(
-		() => !loaded && errorCount < 5,
-		next => {
-			self.getNetwork((err, network) => {
-				if (err) {
-					errorCount += 1;
-					return next();
-				}
-				const peer =
-					network.peers[Math.floor(Math.random() * network.peers.length)];
-				let lastBlock = modules.blocks.lastBlock.get();
-
-				function loadBlocks() {
-					__private.blocksToSync = peer.height;
-
-					modules.blocks.process.loadBlocksFromPeer(
-						peer,
-						(err, lastValidBlock) => {
+		() => !loaded && failedAttemptsToLoad < 5,
+		whilstCb => {
+			async.waterfall(
+				[
+					waterfallCb => {
+						self.getRandomPeerFromNetwork((err, peer) => {
 							if (err) {
-								library.logger.error(err.toString());
+								failedAttemptsToLoad += 1;
 								library.logger.error(
-									`Failed to load blocks from: ${peer.string}`
+									'Failed to get random peer from network',
+									err.toString()
 								);
-								errorCount += 1;
+								return whilstCb();
 							}
-							loaded = lastValidBlock.id === lastBlock.id;
-							lastBlock = null;
-							lastValidBlock = null;
-							next();
+							__private.blocksToSync = peer.height;
+							const lastBlock = modules.blocks.lastBlock.get();
+							library.logger.info(
+								`Looking for common block with: ${peer.string}`
```

Comment:
`Looking from common block from: peer`

Reply:
I don't think it is correct the way you propose. Please, try again or resolve this conversation.